src/decrypt.rs

```diff
@@ -11,42 +11,29 @@ use bytes::Bytes;
 use itertools::Itertools;
 use rayon::prelude::*;
 use std::io::Cursor;
-use std::sync::Arc;
 use xor_name::XorName;
 
-pub fn decrypt(src_hashes: Vec<XorName>, encrypted_chunks: Vec<EncryptedChunk>) -> Result<Bytes> {
-    let src_hashes = Arc::new(src_hashes);
+pub fn decrypt(src_hashes: Vec<XorName>, encrypted_chunks: &[&EncryptedChunk]) -> Result<Bytes> {
     let num_chunks = encrypted_chunks.len();
     let cpus = num_cpus::get();
     let batch_size = usize::max(1, (num_chunks as f64 / cpus as f64).ceil() as usize);
 
     let raw_chunks: Vec<(usize, Bytes)> = encrypted_chunks
         .chunks(batch_size)
         .par_bridge()
-        .map(|batch| DecryptionBatch {
-            jobs: batch
-                .iter()
-                .map(|c| DecryptionJob {
-                    index: c.index,
-                    encrypted_content: c.content.clone(),
-                    src_hashes: src_hashes.clone(),
-                })
-                .collect_vec(),
-        })
         .map(|batch| {
             batch
-                .jobs
                 .par_iter()
                 .map(|c| {
-                    Ok::<(usize, Bytes), Error>((
-                        c.index,
-                        decrypt_chunk(c.index, c.encrypted_content.clone(), c.src_hashes.as_ref())?,
-                    ))
+                    // we can pass &src_hashes since Rayon uses scopes under the hood which guarantees that threads are
+                    // joined before src_hashes goes out of scope
+                    let bytes = decrypt_chunk(c.index, c.content.clone(), &src_hashes)?;
```

Comment:
does this not work also? I dont see where `c` is being used after this?
```suggestion
                    let bytes = decrypt_chunk(c.index, c.content, &src_hashes)?;
```